File: 08-11-2024_Minimum repeat to make substring.java
public class Solution {
    public int minRepeats(String s1, String s2) {
        int minRepeats = (s2.length() + s1.length() - 1) / s1.length();
        StringBuilder repeatedS1 = new StringBuilder();
        for (int i = 0; i < minRepeats; i++) {
            repeatedS1.append(s1);
        }
        if (repeatedS1.indexOf(s2) != -1) {
            return minRepeats;
        }
        repeatedS1.append(s1);
        if (repeatedS1.indexOf(s2) != -1) {
            return minRepeats + 1;
        }
        return -1;
    }
}

File: 08-11-2024_Minimum repeat to make substring.py
def min_repeats(s1: str, s2: str) -> int:
    min_repeats = (len(s2) + len(s1) - 1) // len(s1)
    repeated_s1 = s1 * min_repeats
    if s2 in repeated_s1:
        return min_repeats
    repeated_s1 += s1
    if s2 in repeated_s1:
        return min_repeats + 1
    return -1
